### ISSUE TITLE
Store the library version in the cmake properties file?

### ISSUE THREAD

**Issue by emmenlau** [BlueBrain/HighFive#161]
HighFive generates nice cmake targets for downstream projects. But as far as I could see, it does not currently generate version information. Would that be possible to add?

Typically something like the following is a good start. The main configurable flag is the version `COMPATIBILITY` between releases...
```
include(CMakePackageConfigHelpers)
write_basic_package_version_file(
    cmake/${PROJECT_NAME}ConfigVersion.cmake
    VERSION ${PROJECT_VERSION}
    COMPATIBILITY AnyNewerVersion)
```

**Comment by emmenlau** [comment]
Implemented in #162, please review.

**Comment by emmenlau** [comment]
Thanks for merging!
